zlib-rs/src/inflate.rs

```diff
@@ -1635,6 +1947,17 @@ pub unsafe fn copy(dest: *mut z_stream, source: &InflateStream) -> ReturnCode {
     let writer: MaybeUninit<ReadBuf> =
         unsafe { std::ptr::read(&state.writer as *const _ as *const MaybeUninit<ReadBuf>) };
 
+    // Deep copy the Gzip header.
+    // TODO: is that what we want here?
+    let mut head_copy;
+    let head_opt = match &state.head {
+      Some(head) => {
+        head_copy = *(*head);
+        Some(&mut head_copy)
+      },
+      None => None,
+    };
```

Comment:
I don't think so. This is why this function is rather nasty: the `InflateStream` contains mutable references that we cannot just copy. That is why this `addr_of_mut!` stuff is needed in this function. We´ll need something similar here. 

we _cannot_  duplicate the reference, hence the copied value cannot be considered initialized.